CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ set(NO_AFFINITY 1)
 endif()
 option(CPP_THREAD_SAFETY_TEST "Run a massively parallel DGEMM test to confirm thread safety of the library (requires OpenMP and about 1.3GB of RAM)" OFF)
 option(CPP_THREAD_SAFETY_GEMV "Run a massively parallel DGEMV test to confirm thread safety of the library (requires OpenMP)" OFF)
+option(BUILD_STATIC_LIBS "Build static library" ON)
```

Comment:
Can you make this depend on `BUILD_SHARED_LIBS` and make it be OFF if `BUILD_SHARED_LIBS=ON` by default? That way, this can be backward compatible.